src/js/hooks/list-pages/inbound/useInboundListTableData.jsx

```diff
@@ -0,0 +1,123 @@
+import { useEffect } from 'react';
+
+import _ from 'lodash';
+import { confirmAlert } from 'react-confirm-alert';
+import { getTranslate } from 'react-localize-redux';
+import { useDispatch, useSelector } from 'react-redux';
+import Alert from 'react-s-alert';
+
+import { fetchShipmentStatusCodes, hideSpinner, showSpinner } from 'actions';
+import stockMovementApi from 'api/services/StockMovementApi';
+import { STOCK_MOVEMENT_API, STOCK_MOVEMENT_INCOMING_ITEMS } from 'api/urls';
+import useTableData from 'hooks/list-pages/useTableData';
+import exportFileFromAPI from 'utils/file-download-util';
+import { translateWithDefaultMessage } from 'utils/Translate';
+
+const useInboundListTableData = (filterParams) => {
+  const messageId = 'react.stockMovement.inbound.fetching.error';
+  const defaultMessage = 'Unable to fetch inbound movements';
+  const getParams = (offset, currentLocation, state, sortingParams) => _.omitBy({
+    ...filterParams,
+    offset: `${offset}`,
+    max: `${state.pageSize}`,
+    receiptStatusCode: filterParams.receiptStatusCode &&
+              filterParams.receiptStatusCode?.map(({ id }) => id),
+    origin: filterParams?.origin?.id,
+    destination: filterParams?.destination?.id,
+    requestedBy: filterParams.requestedBy?.id,
+    createdBy: filterParams.createdBy?.id,
+    updatedBy: filterParams.updatedBy?.id,
+    ...sortingParams,
+  }, (value) => {
+    if (typeof value === 'object' && _.isEmpty(value)) return true;
+    return !value;
```

Comment:
I think that this function's arguments should be as `( { offset, currentLocation, state, sortingParams })`, because having so many arguments it might be difficult to follow the order of arguments. Making this change, you'll allow yourself not to pass them in the order that you have now.
Moreover I think that for better readability, it would be good to destructure `filterParams`, having:
```
const { receiptStatusCode, origin, destination, requestedBy, createdBy, updatedBy } = filterParams
```
and proceed with that, so we have:
```
origin: origin?.id
```